### PR TITLE
Replace joi validation with mongoose validation

### PR DESCRIPTION
**Pull request by shiv-chan**
I've replaced `@hapi/joi` validator with a mongoose validator.

This is because we can set a custom error message with a mongoose validator.  [Here's](https://mongoosejs.com/docs/validation.html#custom-error-messages) a detail.

Implemented this custom error messages feature by referring to [this video series](https://www.youtube.com/watch?v=DmrjFKTLOYo&list=PL4cUxeGkcC9iqqESP8335DA5cRFp8loyp) No.5 to No.7.

Also, I added a validator to check if the email is valid and if the password meets the requirement.

The password requirement is... 
1. at least 6 characters
2. includes at least one lowercase
3. includes at least one number.

### CODE REVIEW

File: server/models/userModels.js [hunk 1]
```diff
@@ -1,14 +1,36 @@
 import mongoose from 'mongoose';
 import { v4 as uuidv4 } from 'uuid';
+import validator from 'validator';
```
Comment (miyabitanimchi) [inline server/models/userModels.js:3]: validator is imported here and used on userModels.js, but it seems like validator is not installed? (I didn't see it on package.json so I installed it to test your code)

**Review by miyabitanimchi — COMMENTED** [review]
Thank you for your work! I tested the functions on postman and checked the validation worked correctly. I just left one comment, so please check when you have time, thank you!

File: server/models/userModels.js (hunk 1 above)
Comment (shiv-chan) [reply server/models/userModels.js:3]: It seems it's working without installing the package...wierd!
I installed the package anyway.

Thanks for checking 🙏

**Review by miyabitanimchi — APPROVED** [review]

**Review by Dannyboy222 — APPROVED** [review]
👍

**Pull request merged**